spec/firebase_app_distribution_action_spec.rb

```diff
@@ -1,141 +1,214 @@
 describe Fastlane::Actions::FirebaseAppDistributionAction do
   let(:fake_file) { StringIO.new }
-
-  before do
+  let(:temp_connection) { double("Connection") }
+  # These examples are copied directly from API responses. Maybe we should move them to another file?
+  let(:fake_get_app_response_valid) { double("Response", status: 200, body: { projectNumber: "1", appId: "1", platform: "android", bundleId: "1", contactEmail: "Hello@world.com" }) }
+  let(:fake_get_app_response_no_email) { double("Response", status: 200, body: { projectNumber: "1", appId: "1", platform: "android", bundleId: "1", contactEmail: "" }) }
+  let(:fake_get_app_response_invalid_app_id) { double("Response", status: 400, body: { error: { code: 400, message: "Request contains an invalid argument.", status: "INVALID_ARGUMENT" } }) }
+  let(:fake_get_app_response_permission_denied) { double("Response", status: 403, body: { error: { code: 403, message: "The caller does not have permission", status: "PERMISSION_DENIED" } }) }
+  let(:fake_upload_binary_response_valid) { double("Response", status: 202, body: { token: "projects/project_id/apps/app_id/releases/-/binaries/binary_hash" }) }
+  let(:fake_upload_status_response_success) { double("Response", status: 200, body: { status: "SUCCESS", release: { distributedAt: "1970-01-01T00:00:00Z", id: "status_id", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z", displayVersion: "1.0", buildVersion: "1" } }) }
+  let(:fake_upload_status_response_invalid_app_token) { double("Response", status: 200, body: { status: "ERROR", message: "Update your Gradle plugin to >=2.0.0 or your Firebase CLI to >=8.4.1 to use the faster, more reliable upload pipeline.", release: { distributedAt: "1970-01-01T00:00:00Z", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z" } }) }
+  let(:fake_binary) { double("Binary") }
+
+  before(:each) do
+    allow(Fastlane::Actions::FirebaseAppDistributionAction).to receive(:connection).and_return(temp_connection)
+    allow(fake_binary).to receive(:read).and_return("Hello World")
+
+    allow(File).to receive(:open).and_return(fake_binary)
     allow(Tempfile).to receive(:new).and_return(fake_file)
```

Comment:
By the way I don't think you need Tempfile, do you? I'm guessing this was something for the old implementation/tests.